ibis/pyspark/compiler.py

```diff
@@ -754,3 +822,139 @@ def compile_join(t, expr, scope, how):
     predicates = t.translate(op.predicates[0], scope)
 
     return left_df.join(right_df, predicates, how)
+
+
+@compiles(ops.WindowOp)
+def compile_window_op(t, expr, scope, **kwargs):
+    op = expr.op()
+    window = op.window
+    operand = op.expr
+
+    group_by = window._group_by
+    grouping_keys = [
+        key_op.name
+        if isinstance(key_op, ops.TableColumn)
+        else compile_with_scope(
+            t, key, scope
+        )
+        for key, key_op in zip(
+            group_by, map(operator.methodcaller('op'), group_by)
+        )
+    ]
+
+    order_by = window._order_by
+    ordering_keys = [
+        key.to_expr().get_name()
+        for key in map(operator.methodcaller('op'), order_by)
+    ]
+
+    pyspark_window = Window.partitionBy(grouping_keys).orderBy(ordering_keys)
+    result = t.translate(operand, scope, window=pyspark_window)
+
+    return result
+
+
+@compiles(ops.Lag)
+def compile_lag(t, expr, scope, *, window, **kwargs):
+    op = expr.op()
+
+    src_column = t.translate(op.arg, scope)
+    default = op.default.op().value if op.default is not None else op.default
+    offset = op.offset.op().value if op.offset is not None else op.offset
+
+    if offset:
+        return F.lag(src_column, count=offset, default=default).over(window)
+    else:
+        return F.lag(src_column, default=default).over(window)
+
+
+@compiles(ops.Lead)
+def compile_lead(t, expr, scope, *, window, **kwargs):
```

Comment:
since this and `Lag` are so similar, can we factor them into a single function that takes the pyspark function as a parameter?

Reply:
Yep, done.